src/samplers/rezolus/rusage/mod.rs

```diff
@@ -0,0 +1,129 @@
+use super::stats::*;
+use super::*;
+use crate::common::Counter;
+use crate::common::Nop;
+
+const S: u64 = 1_000_000_000;
+const US: u64 = 1_000;
+const KB: i64 = 1024;
+
+#[distributed_slice(REZOLUS_SAMPLERS)]
+fn init(config: &Config) -> Box<dyn Sampler> {
+    if let Ok(rusage) = Rusage::new(config) {
+        Box::new(rusage)
+    } else {
+        Box::new(Nop {})
+    }
+}
+
+const NAME: &str = "rezolus_rusage";
+
+pub struct Rusage {
+    prev: Instant,
+    next: Instant,
+    interval: Duration,
+    ru_utime: Counter,
+    ru_stime: Counter,
+}
+
+impl Rusage {
+    pub fn new(config: &Config) -> Result<Self, ()> {
+        // check if sampler should be enabled
+        if !config.enabled(NAME) {
+            return Err(());
+        }
+
+        let now = Instant::now();
+
+        Ok(Self {
+            prev: now,
+            next: now,
+            interval: config.interval(NAME),
+            ru_utime: Counter::new(&RU_UTIME, Some(&RU_UTIME_HISTOGRAM)),
+            ru_stime: Counter::new(&RU_STIME, Some(&RU_STIME_HISTOGRAM)),
+        })
+    }
+}
+
+impl Sampler for Rusage {
+    fn sample(&mut self) {
+        let now = Instant::now();
+
+        if now < self.next {
+            return;
+        }
+
+        let elapsed = (now - self.prev).as_secs_f64();
+
+        self.sample_rusage(elapsed);
+
+        // determine when to sample next
+        let next = self.next + self.interval;
```

Comment:
Any reason not to just do `now + self.interval` in all cases? Seems like this would allow you to run a little bit over consistently until you are just about one interval behind and then do that anyway.

Reply:
That might be reasonable. Deferring that as it applies to all samplers

Reply:
Ack, works for me.